giskard/utils/analytics_collector.py

```diff
@@ -149,6 +146,7 @@ def track(self, event_name, properties=None, meta=None, force=False):
         return self._track(event_name, properties=properties, meta=meta, force=force)
 
     @threaded
+    @analytics_method
```

Comment:
If I remember correctly, this was removed to avoid infinite loops. Can we add a test to ensure that this is not happening  (since `_report_error` has been removed in the decorator, this should be the case) and make sure we will not reintroduce this in the future?

EDIT: maybe the test that is present already covers this case?

Reply:
it was actually the presence of 
```
 _report_error(e, error_type="tracking error")
```
that was causing an infinite loop